.github/workflows/codeql.yml

```diff
@@ -0,0 +1,127 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  # push:
+  #   branches: [ "main", "master" ]
+  schedule:
+    - cron: '0 0 * * *'
```

Comment:
When people are dealing with merge requests, there is a shortage of CI runners. Which timezone does this job run at midnight on? How long has the job taken to complete in your test runs?